crates/solver/src/settlement_submission.rs

```diff
@@ -159,6 +159,10 @@ impl SolutionSubmitter {
                 .iter()
                 .map(|strategy| {
                     async {
+                        // Needed so that the compiler understands that we don't
+                        // want to move `strategy` into the closure but use a
+                        // reference to it.
+                        #[allow(clippy::borrow_deref_ref)]
```

Comment:
This is the one "non-trivial" lint. Specifically:
```
warning: deref on an immutable reference
   --> crates/solver/src/settlement_submission.rs:162:31
    |
162 |                         match &*strategy {
    |                               ^^^^^^^^^^ help: if you would like to reborrow, try removing `&*`: `strategy`
    |
    = note: `#[warn(clippy::borrow_deref_ref)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#borrow_deref_ref
```

The suggestion to remove the `&*` causes the compiler to complain that you need to move `strategy` into the closure. However, adding `move` will also cause `settlement, `network_id` and `account` to move into the closure which is problematic. The alternative solution would be to:
```rust
let (settlement, network_id, account) = (&settlement, &network_id, &account);
async move {
    match strategy { /* ... */ }
    // ...
}
```